### PR TITLE
Parameterize directories used by dotnet-ef

### LINKED ISSUE #5311: CLI Commands: support loading startup from a separate project
Add support for supplying a project as a parameter in `--startup-project` that is different than the current project.

This is explicitly disabled in code. See https://github.com/aspnet/EntityFramework/issues/4577#issuecomment-217496222 for background and https://github.com/aspnet/EntityFramework/pull/5312.

### PR DESCRIPTION
**Pull request by natemcmaster**
Fix #5311 by adding back `--startup-project`.

NB: using `--startup-project` requires that "Microsoft.EntityFrameworkCore.Design" be installed startup project, not the current project. This partially enables class libraries. cref #5320

TODO: parameter order on command line still matters, so this option must be added between "ef" and the next sub command. E.g. `dotnet ef --startup-project ../WebApp1 database update`. This is already tracked in #5188 

This is built under the assumption that on .NET Core CLI, the current working directory is your target project, _and is not necessarily the startup project_. The reverse assumption (that dotnet-ef should always be executed on startup) is not consistent with other .NET Core CLI commands that alter files.

Worth noting: we are still using relative project directories as the value of `--startup-project`. This is to stay consistent with how .NET Core CLI handles preferences to projects. (Example see `dotnet run --help`)

cc @bricelam @divega @rowanmiller

### CODE REVIEW

**Comment by natemcmaster** [comment]
(PS, before I request ask-mode approval, let's consider first that this is the behavior we want.)

**Comment by bricelam** [comment]
Wonderful! It's exactly how I always wanted it to work.

File: src/Microsoft.EntityFrameworkCore.Design/CommandLine/ExecuteCommand.cs [hunk 1]
```diff
@@ -80,12 +75,16 @@ public static CommandLineApplication Create(ref string[] args)
...
+                StartupAssembly = app.Option(StartupAssemblyOptionTemplate + " <ASSEMBLY>",
+                     "The assembly file containing the startup class."),
                 DataDirectory = app.Option(DataDirectoryOptionTemplate + " <DIR>",
-                    "The folder to use as the data directory. If not specified, the runtime output folder is used."),
+                    "The folder used as the data directory."),
                 ProjectDirectory = app.Option(ProjectDirectoryOptionTemplate + " <DIR>",
-                    "The folder to use as the project directory. If not specified, the current working is used."),
+                    "The folder used as the project directory. If not specified, the current working directory is used."),
+                StartupTargetDirectory = app.Option(StartupTargetDirectoryOptionTemplate + " <DIR>",
+                    "The folder used as the base directory for ASP.NET hosting environment. If not specified, the current working directory is used."),
```
Comment (bricelam) [inline src/Microsoft.EntityFrameworkCore.Design/CommandLine/ExecuteCommand.cs]: This isn't ASP.NET-specific. Maybe something like "...base directory for the application."

**Comment by bricelam** [comment]
:shipit:

File: src/Microsoft.EntityFrameworkCore.Design/CommandLine/ExecuteCommand.cs (hunk 1 above)
Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Design/CommandLine/ExecuteCommand.cs]: Can we drop the data directory and merge them into one parameter? I can't think of a scenario in which data directory is different from the app base path.

Comment (bricelam) [inline src/Microsoft.EntityFrameworkCore.Design/CommandLine/ExecuteCommand.cs]: Classic ASP.NET uses `<base>/App_Data`

Comment (bricelam) [inline src/Microsoft.EntityFrameworkCore.Design/CommandLine/ExecuteCommand.cs]: But yes, it's probably OK to do that for now on CLI.

Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Design/CommandLine/ExecuteCommand.cs]: I think I'm going to leave it, as long as there is the possible scenario that data directory may not be the app base path.

**Pull request merged**